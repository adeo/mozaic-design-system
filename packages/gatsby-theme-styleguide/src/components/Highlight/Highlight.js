import React from 'react'
import styled, { css } from 'styled-components'

import {
  ColorInfo100,
  ColorInfo500,
  ColorInfo700,
  ColorWarning100,
  ColorWarning500,
  ColorWarning700,
  ColorGrey100,
  ColorGrey500,
  ColorGrey700,
} from '@mozaic-ds/tokens/build/js/tokens'

const defaultTitles = {
  info: 'Information',
  warning: 'Warning',
  tips: 'Useful tip',
}

const Container = styled.div`
  margin: 2rem 0;
  padding: 1rem 1.5rem 2rem;

  ${({ theme }) =>
    theme === 'info'
      ? css`
          background-color: ${ColorGrey100};
          border-left: solid 8px ${ColorGrey500};
        `
      : theme === 'warning'
      ? css`
          background-color: ${ColorWarning100};
          border-left: solid 8px ${ColorWarning500};
        `
      : theme === 'tips' &&
        css`
          background-color: ${ColorInfo100};
          border-left: solid 8px ${ColorInfo500};
        `}
`

export const Theme = styled.div`
  font-size: 0.875rem;
  line-height: 1;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  font-weight: bold;
  letter-spacing: 0.05em;
  text-transform: uppercase;

  ${({ theme }) =>
    theme === 'info'
      ? css`
          color: ${ColorGrey700};
        `
      : theme === 'warning'
      ? css`
          color: ${ColorWarning700};
        `
      : theme === 'tips' &&
        css`
          color: ${ColorInfo700};
        `}
`

const Content = styled.div`
  margin-top: 0.5rem;

  > :first-child {
    margin-top: 0;
  }
  > :last-child {
    margin-bottom: 0;
  }

  h1,
  h2,
  h3,
  h4 {
    margin-bottom: 0.5em;

    ${({ theme }) =>
      theme === 'info'
        ? css`
            color: ${ColorGrey700};
          `
        : theme === 'warning'
        ? css`
            color: ${ColorWarning700};
          `
        : theme === 'tips' &&
          css`
            color: ${ColorInfo700};
          `}
  }
`

const Highlight = ({ children, theme = 'info', title }) => (
  <Container theme={theme}>
    <Theme theme={theme}>{title ? title : defaultTitles[theme]}</Theme>
    <Content theme={theme}>{children}</Content>
  </Container>
)

export default Highlight
