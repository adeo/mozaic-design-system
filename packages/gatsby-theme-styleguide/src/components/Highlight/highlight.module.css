.container {
  margin: 2rem 0;
  padding: 1rem 1.5rem 2rem;
  background-color: var(--background, transparent);
  border-left: solid 8px var(--border, transparent);
}

.title {
  align-items: center;
  color: var(--color, black);
  display: inline-flex;
  font-size: 0.875rem;
  font-weight: bold;
  justify-content: center;
  letter-spacing: 0.05em;
  line-height: 1;
  text-transform: uppercase;
}

.content {
  margin-top: 0.5rem;
}

.content > :first-child {
  margin-top: 0;
}

.content > :last-child {
  margin-bottom: 0;
}
