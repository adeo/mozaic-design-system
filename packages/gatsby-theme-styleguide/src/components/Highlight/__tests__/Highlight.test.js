import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Highlight, { Theme } from '../Highlight'

describe(`<Highlight>`, () => {
  it(`should render with info theme if no theme is provided`, () => {
    const tree = renderer.create(<Highlight>Hello</Highlight>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`should display default info title if no title props is provided`, () => {
    const tree = shallow(<Highlight>Hello</Highlight>)
    expect(tree.find(Theme).text()).toBe('Information')
  })

  it(`should render with warning theme`, () => {
    const tree = renderer
      .create(<Highlight theme="warning">Hello</Highlight>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`should display warning title if warning theme props is provided`, () => {
    const tree = shallow(<Highlight theme="warning">Hello</Highlight>)
    expect(tree.find(Theme).text()).toBe('Warning')
  })

  it(`should render with info theme`, () => {
    const tree = renderer
      .create(<Highlight theme="info">Hello</Highlight>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`should display info title if info theme props is provided`, () => {
    const tree = shallow(<Highlight theme="info">Hello</Highlight>)
    expect(tree.find(Theme).text()).toBe('Information')
  })

  it(`should render with tips theme`, () => {
    const tree = renderer
      .create(<Highlight theme="tips">Hello</Highlight>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`should display default tips title if no props is provided`, () => {
    const tree = shallow(<Highlight theme="tips">Hello</Highlight>)
    expect(tree.find(Theme).text()).toBe('Useful tip')
  })

  it(`should overide title if title prop is provided`, () => {
    const tree = shallow(
      <Highlight theme="tips" title="title">
        Hello
      </Highlight>
    )
    expect(tree.find(Theme).text()).toBe('title')
  })
})
