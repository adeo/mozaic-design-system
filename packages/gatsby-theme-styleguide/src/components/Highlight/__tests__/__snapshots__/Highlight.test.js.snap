// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight> should render with info theme 1`] = `
.c0 {
  margin: 2rem 0;
  padding: 1rem 1.5rem 2rem;
  background-color: #eeeef0;
  border-left: solid 8px #887f87;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  color: #554f52;
}

.c2 {
  margin-top: 0.5rem;
}

.c2 > :first-child {
  margin-top: 0;
}

.c2 > :last-child {
  margin-bottom: 0;
}

.c2 h1,
.c2 h2,
.c2 h3,
.c2 h4 {
  margin-bottom: 0.5em;
  color: #554f52;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Information
  </div>
  <div
    className="c2"
  >
    Hello
  </div>
</div>
`;

exports[`<Highlight> should render with info theme if no theme is provided 1`] = `
.c0 {
  margin: 2rem 0;
  padding: 1rem 1.5rem 2rem;
  background-color: #eeeef0;
  border-left: solid 8px #887f87;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  color: #554f52;
}

.c2 {
  margin-top: 0.5rem;
}

.c2 > :first-child {
  margin-top: 0;
}

.c2 > :last-child {
  margin-bottom: 0;
}

.c2 h1,
.c2 h2,
.c2 h3,
.c2 h4 {
  margin-bottom: 0.5em;
  color: #554f52;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Information
  </div>
  <div
    className="c2"
  >
    Hello
  </div>
</div>
`;

exports[`<Highlight> should render with tips theme 1`] = `
.c0 {
  margin: 2rem 0;
  padding: 1rem 1.5rem 2rem;
  background-color: #ecfcfd;
  border-left: solid 8px #25a8d0;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  color: #1e5f82;
}

.c2 {
  margin-top: 0.5rem;
}

.c2 > :first-child {
  margin-top: 0;
}

.c2 > :last-child {
  margin-bottom: 0;
}

.c2 h1,
.c2 h2,
.c2 h3,
.c2 h4 {
  margin-bottom: 0.5em;
  color: #1e5f82;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Useful tip
  </div>
  <div
    className="c2"
  >
    Hello
  </div>
</div>
`;

exports[`<Highlight> should render with warning theme 1`] = `
.c0 {
  margin: 2rem 0;
  padding: 1rem 1.5rem 2rem;
  background-color: #fff5eb;
  border-left: solid 8px #fc961e;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  color: #974f20;
}

.c2 {
  margin-top: 0.5rem;
}

.c2 > :first-child {
  margin-top: 0;
}

.c2 > :last-child {
  margin-bottom: 0;
}

.c2 h1,
.c2 h2,
.c2 h3,
.c2 h4 {
  margin-bottom: 0.5em;
  color: #974f20;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Warning
  </div>
  <div
    className="c2"
  >
    Hello
  </div>
</div>
`;
